hazelcast/src/main/java/com/hazelcast/scheduledexecutor/impl/ScheduledExecutorContainer.java

```diff
@@ -222,11 +223,15 @@ public ScheduledTaskHandler offprintHandler(String taskName) {
     public void promoteSuspended() {
         for (ScheduledTaskDescriptor descriptor : tasks.values()) {
             try {
-                log(FINEST, descriptor.getDefinition().getName(), "Attempting promotion");
+                final String taskName = descriptor.getDefinition().getName();
+
                 if (descriptor.shouldSchedule()) {
+                    log(FINEST, taskName, "Promoting");
                     doSchedule(descriptor);
+                } else {
+                    log(FINEST, taskName, "Notifying ready result");
```

Comment:
So if I'm understanding this fix correctly; in case of a split brain healing, all tasks that should not be rescheduled are completed? 

How come that the task completion needs to be triggered because of a merge? If the task was ready, why didn't it complete in the first place?

Reply:
The hypothesis behind this fix is that we schedule a task (at a partition that belongs to Hazelcast A), got a Future at hand (from the HazelcastInstance C) and then we have a split-brain. [A, B] & [C]. From C (where the Future proxy lives) we trigger `.get()` (assuming that C is now promoting its own replica copy). Before this happens, and before the task finishes, we heal the cluster, and the tasks get merged. We keep the original task from A, which was completed, earlier, however, the Future.get is still parked waiting for that notification it never received.